pennylane/math/multi_dispatch.py

```diff
@@ -772,9 +772,15 @@ def add(*args, like=None, **kwargs):
     try:
         return np.add(*args, **kwargs)
     except TypeError:
-        # catch arg1 = torch, arg2=numpy error
-        # works fine with opposite order
-        return np.add(args[1], args[0], *args[2:], **kwargs)
+        # In autoray 0.6.5, np.add dispatches to torch instead of
+        # numpy if one parameter is a torch tensor and the other is
+        # a numpy array. torch.add raises an Exception if one of the
+        # arguments is a numpy array, so here we cast both arguments
+        # to be tensors.
+        dev = getattr(args[0], "device", None) or getattr(args[1], "device")
+        arg0 = np.asarray(args[0], device=dev, like=like)
+        arg1 = np.asarray(args[1], device=dev, like=like)
+        return np.add(arg0, arg1, *args[2:], **kwargs)

```

Comment:
could we check `like="torch"` above to avoid having a try-except block?

Reply:
If both arguments are torch tensors then this casting is unnecessary, so I'd prefer it this way.

Reply:
I second this, also because this _only_ works with torch. Example:
```pycon
>>> np.asarray(1.1, device=None, like="jax")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/matthews/.pyenv/versions/pl/lib/python3.9/site-packages/autoray/autoray.py", line 80, in do
    return get_lib_fn(backend, fn)(*args, **kwargs)
TypeError: asarray() got an unexpected keyword argument 'device'
```

Reply:
I'm just gerenally skeptical of exceptions as control flow and would prefer to avoid it if possible.

Reply:
if you don't want to do this extra casting, you can do `if like == torch and get_interface(args[1]) != "torch"`. Generally, when you know _exactly_ what causes the issue, I think you should just check for exactly that instead of a try-except. Plus, if something else goes wrong, I want it to fail so I can know what I missed, and fix it!